src/adapters/eplanning.js

```diff
@@ -0,0 +1,53 @@
+var bidfactory = require('src/bidfactory.js');
+var bidmanager = require('src/bidmanager.js');
+var adloader = require('src/adloader.js');
+
+var EPlanningAdapter = function EPlanningAdapter() {
+
+  var ISV_DEFAULT = "aklc.img.e-planning.net";
+
+  window.pbjs = window.pbjs || {};
+  window.pbjs.processEPlanningResponse = function(response) {
+    var bids, bidObject, i;
+    if (response) {
+      bids = response.bids;
+      for (i = 0; i < bids.length; i++) {
+        if (bids[i].ad) {
+          bidObject = getBidObject(bids[i]);
+          bidmanager.addBidResponse(bids[i].placementCode, bidObject);
+        } else {
+          bidObject = bidfactory.createBid(2);
+          bidObject.bidderCode = 'eplanning';
+          bidmanager.addBidResponse(bids[i].placementCode, bidObject);
+        }
+      }
+    }
+  };
+
+  function getBidObject(bid) {
+    var bidObject = bidfactory.createBid(1), i;
+    bidObject.bidderCode = 'eplanning';
+    for (i in bid.ad) {
+      if (bid.ad.hasOwnProperty(i)) {
+        bidObject[i] = bid.ad[i];
+      }
+    }
+    return bidObject;
+  }
+
+  function _callBids(params) {
+    var scriptURL = "http://" + ((params && params.bids && params.bids.length && params.bids[0].params && params.bids[0].params.isv) ? params.bids[0].params.isv : ISV_DEFAULT) + "/layers/hbpb.js";
+    adloader.loadScript(scriptURL, function() {
+      if (window.hbpb) {
+        window.hbpb.call(params);
+      }
```

Comment:
@dbemiller yes, the adapter loads an external script hosted by us which do the request to the server and get the bids. This script initilializes with the hbpb.call() function, that's why it is called in the callback. It is true that it could be faulty if someone creates a global variable with the same name, but not less faulty than someone creating a window.googletag = function() { return 1; }, which could happen too.
This code is in an external script as we would like to give this level of features to our publishers. They are the ones that will have a "ci" parameter (clientId) and we are going to bid for, so it won't involve other publishers who do not have an agreement with us. Our bidder (as many others) can't be used if you do not have a client identifier. We provide a 24/7 tech support to all of our publishers to solve this kind of trouble (if it happens).